### DISCUSSION TITLE
Marginal / fuzzy time tracking

### DISCUSSION THREAD

**Discussion by acikek** [jotaen/klog#47] [Feature ideas & discussions]
Sometimes you want something done in a certain amount of time, sort of like a deadline but in a certain time constraint. Other times, you know that a task could take more than the time you expect it could. It would be nice if there was a feature for this.

As `<` and `>` are already in use for day shifting, here's what I propose:

```
2021-03-26
  8h another day at my #job
  ?30m lunch #break, I gotta get my report done!
  1h? watching #football
```
Alternatively, it could be `>1h` or `30m<`, but that could cause issues with the current day shifting syntax. I think either one is pleasing to the eye.

As for `klog total`ing this record, these could be ignored as with open-ended time ranges, or (and I think this is better) they could be accounted for by denoting "or more" (or "or less" by a certain margin).

For example, the record above would be totaled as such:
```
Total: at least 9h or less by ~30m
(In 1 record)
```
Typing that out, it feels clunky. Perhaps there's a better, more intuitive method of totaling that. Overall, though, I think this would be a beneficial addition.

**Comment by jotaen** [comment]
Interesting use-case. I’m not sure though I have fully understood the idea behind it, maybe you can clarify this a bit more?

Would you want this for planning out a day in advance, as in setting up placeholders and then filling them later on with the actual times once you around to them? I guess this would be a more fine-granular version of should-total times, wouldn’t it? Or is this more about introducing a concept of “fuzziness” or precision for the tracking?

**Comment by acikek** [comment]
Fuzziness is one aspect of it, yes. While most tasks go a little off-track in the time they're supposed to take, this is more to exemplify an aspect of a day.

I am a very uncertain person. I have a hard time making up my mind. I also get very sidetracked with a lot of the things I do. Personally, I like to plan out my day, but I can see how this could be used for other purposes. This just gives the end-user another method to track their time, recognizing that tasks and the time they take are not static.

Now that I think about it, this might be better titled "marginal" time trackers, or something.
